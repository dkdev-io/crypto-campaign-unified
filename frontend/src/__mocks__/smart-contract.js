import { vi } from 'vitest';

export const processContribution = vi.fn().mockResolvedValue({
  success: true,
  transactionHash: '0xabc123...',
  blockNumber: 18500000,
  gasUsed: '21000',
});

export default {
  processContribution,
};
