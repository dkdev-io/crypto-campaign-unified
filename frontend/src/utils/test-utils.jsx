import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { vi } from 'vitest';

// Test data factories
export const createMockCampaign = (overrides = {}) => ({
  id: 'test-campaign-1',
  campaign_name: 'Test Campaign',
  email: 'test@example.com',
  candidate_name: 'John Doe',
  theme_color: '#2a2a72',
  suggested_amounts: [25, 50, 100, 250],
  max_donation_limit: 3300,
  website: 'https://test.com',
  wallet_address: '0x123...',
  created_at: '2024-01-01T00:00:00.000Z',
  ...overrides
});

export const createMockDonorData = (overrides = {}) => ({
  fullName: 'Jane Smith',
  email: 'jane@example.com',
  phone: '555-123-4567',
  street: '123 Main St',
  city: 'Anytown',
  state: 'CA',
  zip: '12345',
  employer: 'Test Corp',
  occupation: 'Engineer',
  amount: '100',
  ...overrides
});

export const createMockWalletInfo = (overrides = {}) => ({
  isConnected: true,
  account: '0x742d35Cc6C4C5e5D0E9dF5F9B6A5e8a2c1B9D8e7',
  balance: '1.5678',
  network: {
    name: 'Ethereum Mainnet',
    chainId: '0x1'
  },
  contributorInfo: {
    isKYCVerified: true,
    cumulativeAmount: 0.5,
    remainingCapacity: 1.0
  },
  ...overrides
});

export const createMockFormSubmission = (overrides = {}) => ({
  id: 'submission-1',
  campaign_id: 'test-campaign-1',
  donor_full_name: 'Jane Smith',
  donor_email: 'jane@example.com',
  amount_usd: 100,
  transaction_hash: '0xabc123...',
  status: 'confirmed',
  created_at: '2024-01-01T12:00:00.000Z',
  ...overrides
});

// Custom render function with providers
export const render = (ui, {
  route = '/',
  ...renderOptions
} = {}) => {
  // Set initial URL for router
  window.history.pushState({}, 'Test page', route);
  
  const Wrapper = ({ children }) => (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// Utility to wait for async operations
export const waitForAsync = () => new Promise(resolve => setTimeout(resolve, 0));

// Mock event creators
export const createMockEvent = (overrides = {}) => ({
  preventDefault: vi.fn(),
  stopPropagation: vi.fn(),
  target: {
    value: 'test-value',
    name: 'test-input',
    checked: false,
    ...overrides.target
  },
  currentTarget: {
    value: 'test-value',
    ...overrides.currentTarget
  },
  ...overrides
});

export const createChangeEvent = (value, name = 'test-input') => 
  createMockEvent({
    target: { value, name },
    currentTarget: { value }
  });

export const createClickEvent = (overrides = {}) =>
  createMockEvent({
    type: 'click',
    ...overrides
  });

export const createSubmitEvent = (overrides = {}) =>
  createMockEvent({
    type: 'submit',
    ...overrides
  });

// Form validation helpers
export const fillFormFields = async (user, formData) => {
  const fields = Object.entries(formData);
  
  for (const [fieldName, value] of fields) {
    const input = document.querySelector(`input[name="${fieldName}"], select[name="${fieldName}"], textarea[name="${fieldName}"]`);
    if (input) {
      if (input.type === 'checkbox' || input.type === 'radio') {
        if (value) {
          await user.click(input);
        }
      } else {
        await user.type(input, String(value));
      }
    }
  }
};

// Assertion helpers
export const expectFormField = (container, fieldName, expectedValue = null) => {
  const field = container.querySelector(`[name="${fieldName}"]`);
  expect(field).toBeInTheDocument();
  
  if (expectedValue !== null) {
    if (field.type === 'checkbox' || field.type === 'radio') {
      expect(field.checked).toBe(Boolean(expectedValue));
    } else {
      expect(field.value).toBe(String(expectedValue));
    }
  }
  
  return field;
};

export const expectErrorMessage = (container, message) => {
  const errorElement = container.querySelector('[style*="background: #f8d7da"], [style*="color: #721c24"], .error-message');
  expect(errorElement).toBeInTheDocument();
  if (message) {
    expect(errorElement).toHaveTextContent(message);
  }
};

export const expectSuccessMessage = (container, message) => {
  const successElement = container.querySelector('[style*="background: #d4edda"], [style*="color: #155724"], .success-message');
  expect(successElement).toBeInTheDocument();
  if (message) {
    expect(successElement).toHaveTextContent(message);
  }
};

// Mock HTTP responses
export const mockApiResponse = (data, options = {}) => {
  const {
    status = 200,
    ok = true,
    delay = 0
  } = options;
  
  return new Promise(resolve => {
    setTimeout(() => {
      resolve({
        ok,
        status,
        statusText: ok ? 'OK' : 'Error',
        json: () => Promise.resolve(data),
        text: () => Promise.resolve(JSON.stringify(data))
      });
    }, delay);
  });
};

// Component test helpers
export const getByDataTestId = (container, testId) => {
  return container.querySelector(`[data-testid="${testId}"]`);
};

export const getAllByDataTestId = (container, testId) => {
  return container.querySelectorAll(`[data-testid="${testId}"]`);
};

// Form interaction patterns
export const submitForm = async (user, form) => {
  const submitButton = form.querySelector('button[type="submit"], input[type="submit"]');
  if (submitButton) {
    await user.click(submitButton);
  } else {
    // Submit via form if no submit button found
    await user.submit(form);
  }
};

export const selectSuggestedAmount = async (user, container, amount) => {
  const amountButton = container.querySelector(`button[type="button"]:has-text("$${amount}")`);
  if (amountButton) {
    await user.click(amountButton);
  }
  return amountButton;
};

// Wallet interaction helpers
export const connectWallet = async (user, container) => {
  const connectButton = container.querySelector('button:has-text("Connect"), button:has-text("MetaMask")');
  if (connectButton) {
    await user.click(connectButton);
  }
  return connectButton;
};

export const disconnectWallet = async (user, container) => {
  const disconnectButton = container.querySelector('button:has-text("Disconnect")');
  if (disconnectButton) {
    await user.click(disconnectButton);
  }
  return disconnectButton;
};

// Time utilities for testing
export const mockDateNow = (timestamp) => {
  const dateSpy = vi.spyOn(Date, 'now');
  dateSpy.mockReturnValue(timestamp);
  return dateSpy;
};

export const mockDate = (dateString) => {
  const mockDate = new Date(dateString);
  const dateSpy = vi.spyOn(global, 'Date');
  dateSpy.mockImplementation(() => mockDate);
  return dateSpy;
};

// Export everything for easy importing
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { vi } from 'vitest';