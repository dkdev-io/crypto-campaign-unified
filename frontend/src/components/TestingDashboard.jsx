import React, { useState } from 'react';
import { processContribution } from '../lib/smart-contract';

const TestingDashboard = () => {
  const [testResults, setTestResults] = useState([]);
  const [isRunning, setIsRunning] = useState(false);
  const [testForm, setTestForm] = useState({
    amount: '',
    walletAddress: '',
    firstName: '',
    lastName: '',
    email: '',
    acknowledgmentSigned: false
  });

  const runTest = async () => {
    if (!testForm.amount || !testForm.walletAddress || !testForm.firstName || !testForm.lastName || !testForm.email) {
      alert('Please fill in all required fields');
      return;
    }
<<<<<<< HEAD
    
    if (!testForm.acknowledgmentSigned) {
      alert('Please check the FEC compliance acknowledgment');
      return;
    }
=======
>>>>>>> ed641141884f9405daab140d322b712728df23bf

    setIsRunning(true);
    console.log('üî¨ Running smart contract test with your data...');
    
    try {
<<<<<<< HEAD
      // Ensure all fields are properly passed
      const testData = {
        ...testForm,
        contributorWallet: testForm.walletAddress // Also include this field name
      };
      const result = await processContribution(testData, 'test-campaign-123');
=======
      const result = await processContribution(testForm, 'test-campaign-123');
>>>>>>> ed641141884f9405daab140d322b712728df23bf
      
      const testResult = {
        input: { ...testForm },
        output: result,
        timestamp: new Date().toLocaleTimeString()
      };
      
      console.log(result.success ? '‚úÖ' : '‚ùå', 'Test result:', result);
      setTestResults(prev => [testResult, ...prev]);
      
    } catch (error) {
      console.log('‚ùå', 'Test ERROR:', error.message);
      
      const testResult = {
        input: { ...testForm },
        output: { success: false, error: error.message },
        timestamp: new Date().toLocaleTimeString()
      };
      
      setTestResults(prev => [testResult, ...prev]);
    }
    
    setIsRunning(false);
  };

  return (
    <div style={{ padding: '2rem', maxWidth: '1200px', margin: '0 auto' }}>
      <div style={{ textAlign: 'center', marginBottom: '2rem' }}>
        <h1 style={{ color: '#2a2a72' }}>üß™ Smart Contract Testing Dashboard</h1>
        <p style={{ color: '#666', fontSize: '1.1rem' }}>
          Test contribution limits without triggering wallet extensions
        </p>
        <div style={{
          background: '#e3f2fd',
          padding: '1rem',
          borderRadius: '8px',
          margin: '1rem 0',
          border: '1px solid #bbdefb'
        }}>
          <strong>Testing Mode Active:</strong> No wallet extensions will be triggered. All transactions are simulated.
        </div>
      </div>

      <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '2rem' }}>
        {/* Test Input Form */}
        <div>
          <h2 style={{ marginBottom: '1rem' }}>Smart Contract Test</h2>
          <div style={{
            background: '#f8f9fa',
            padding: '1rem',
            borderRadius: '8px',
            marginBottom: '1rem'
          }}>
            <h3>Smart Contract Requirements:</h3>
            <ul>
              <li><strong>FEC Individual Limit:</strong> $3,300 per election (cumulative)</li>
              <li><strong>Minimum:</strong> $1</li>
              <li><strong>Complete Info:</strong> All fields required</li>
              <li><strong>Compliance:</strong> FEC acknowledgment checkbox must be signed</li>
              <li><strong>KYC:</strong> Requirements (to be added later)</li>
            </ul>
          </div>
          
          <div style={{ display: 'grid', gap: '1rem', marginBottom: '1rem' }}>
            <div>
              <label style={{ display: 'block', marginBottom: '0.5rem', fontWeight: 'bold' }}>
                Contribution Amount *
              </label>
              <input
                type="number"
                min="0"
                step="0.01"
                value={testForm.amount}
                onChange={(e) => setTestForm({...testForm, amount: e.target.value})}
                placeholder="Enter amount to test"
                style={{
                  width: '100%',
                  padding: '0.75rem',
                  border: '1px solid #ccc',
                  borderRadius: '4px'
                }}
              />
            </div>
            
            <div>
              <label style={{ display: 'block', marginBottom: '0.5rem', fontWeight: 'bold' }}>
                Wallet Address *
              </label>
              <input
                type="text"
                value={testForm.walletAddress}
                onChange={(e) => setTestForm({...testForm, walletAddress: e.target.value})}
                placeholder="Enter wallet address"
                style={{
                  width: '100%',
                  padding: '0.75rem',
                  border: '1px solid #ccc',
                  borderRadius: '4px'
                }}
              />
            </div>
            
            <div style={{ display: 'grid', gridTemplateColumns: '1fr 1fr', gap: '1rem' }}>
              <div>
                <label style={{ display: 'block', marginBottom: '0.5rem', fontWeight: 'bold' }}>
                  First Name *
                </label>
                <input
                  type="text"
                  value={testForm.firstName}
                  onChange={(e) => setTestForm({...testForm, firstName: e.target.value})}
                  placeholder="First name"
                  style={{
                    width: '100%',
                    padding: '0.75rem',
                    border: '1px solid #ccc',
                    borderRadius: '4px'
                  }}
                />
              </div>
              <div>
                <label style={{ display: 'block', marginBottom: '0.5rem', fontWeight: 'bold' }}>
                  Last Name *
                </label>
                <input
                  type="text"
                  value={testForm.lastName}
                  onChange={(e) => setTestForm({...testForm, lastName: e.target.value})}
                  placeholder="Last name"
                  style={{
                    width: '100%',
                    padding: '0.75rem',
                    border: '1px solid #ccc',
                    borderRadius: '4px'
                  }}
                />
              </div>
            </div>
            
            <div>
              <label style={{ display: 'block', marginBottom: '0.5rem', fontWeight: 'bold' }}>
                Email *
              </label>
              <input
                type="email"
                value={testForm.email}
                onChange={(e) => setTestForm({...testForm, email: e.target.value})}
                placeholder="Email address"
                style={{
                  width: '100%',
                  padding: '0.75rem',
                  border: '1px solid #ccc',
                  borderRadius: '4px'
                }}
              />
            </div>
          </div>
          
          <div style={{
            background: '#f0f8ff',
            padding: '1rem',
            borderRadius: '4px',
            marginBottom: '1rem',
            fontSize: '0.9rem'
          }}>
            <label style={{ display: 'flex', alignItems: 'flex-start', gap: '0.5rem' }}>
              <input
                type="checkbox"
                checked={testForm.acknowledgmentSigned}
                onChange={(e) => setTestForm({...testForm, acknowledgmentSigned: e.target.checked})}
                style={{ marginTop: '0.25rem' }}
              />
              <span>
                I certify that I am a U.S. citizen or lawfully admitted permanent resident, 
                this contribution is made from my own funds, I am not a federal contractor, 
                and I am at least 18 years old.
              </span>
            </label>
          </div>
          
          <button
            onClick={runTest}
            disabled={isRunning}
            style={{
              width: '100%',
              padding: '1rem',
              background: isRunning ? '#ccc' : '#6f42c1',
              color: 'white',
              border: 'none',
              borderRadius: '8px',
              fontSize: '1.1rem',
              cursor: isRunning ? 'not-allowed' : 'pointer'
            }}
          >
            {isRunning ? '‚è≥ Testing Smart Contract...' : 'üß™ Test Smart Contract Limits'}
          </button>
        </div>

        {/* Test Results */}
        <div>
          <h2 style={{ marginBottom: '1rem' }}>Test Results</h2>
          <div style={{ display: 'grid', gap: '1rem', maxHeight: '600px', overflowY: 'auto' }}>
            {testResults.length === 0 ? (
              <div style={{
                background: '#f8f9fa',
                padding: '2rem',
                textAlign: 'center',
                borderRadius: '8px',
                color: '#666'
              }}>
                No tests run yet. Fill in the form and click "Test Smart Contract Limits" to test your data.
              </div>
            ) : (
              testResults.map((result, index) => (
                <div
                  key={index}
                  style={{
                    background: result.output.success ? '#d4edda' : '#f8d7da',
                    border: `1px solid ${result.output.success ? '#c3e6cb' : '#f5c6cb'}`,
                    padding: '1rem',
                    borderRadius: '8px'
                  }}
                >
                  <div style={{
                    display: 'flex',
                    justifyContent: 'space-between',
                    alignItems: 'center',
                    marginBottom: '0.5rem'
                  }}>
                    <strong>{result.output.success ? '‚úÖ SUCCESS' : '‚ùå FAILED'}</strong>
                    <small>{result.timestamp}</small>
                  </div>
                  <div style={{ fontSize: '0.9rem' }}>
                    <div><strong>Amount Tested:</strong> ${result.input.amount}</div>
                    <div><strong>Name:</strong> {result.input.firstName} {result.input.lastName}</div>
                    <div><strong>Email:</strong> {result.input.email}</div>
                    <div><strong>Wallet:</strong> {result.input.walletAddress?.substring(0, 20)}...</div>
                    {result.output.success ? (
                      <div style={{ color: '#155724', marginTop: '0.5rem' }}>
                        <div>‚úÖ Transaction Hash: {result.output.transactionHash?.substring(0, 20)}...</div>
                        <div>Block: {result.output.blockNumber}</div>
                        <div>Gas Used: {result.output.gasUsed}</div>
                      </div>
                    ) : (
                      <div style={{ color: '#721c24', marginTop: '0.5rem' }}>
                        ‚ùå Error: {result.output.error}
                      </div>
                    )}
                  </div>
                </div>
              ))
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default TestingDashboard;