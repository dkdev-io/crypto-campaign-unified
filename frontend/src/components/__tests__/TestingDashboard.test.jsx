import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, userEvent, waitFor } from '../../utils/test-utils';
import TestingDashboard from '../TestingDashboard';
import { processContribution } from '../../lib/smart-contract';

// Mock the smart-contract module
vi.mock('../../lib/smart-contract', () => ({
  processContribution: vi.fn(),
}));

describe('TestingDashboard', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    processContribution.mockResolvedValue({
      success: true,
      transactionHash: '0xabc123...',
      blockNumber: 18500000,
      gasUsed: '21000',
    });
  });

  describe('Component Rendering', () => {
    it('renders testing dashboard with correct title and description', () => {
      render(<TestingDashboard />);

      expect(screen.getByText('🧪 Smart Contract Testing Dashboard')).toBeInTheDocument();
      expect(
        screen.getByText(/Test contribution limits without triggering wallet extensions/)
      ).toBeInTheDocument();
      expect(screen.getByText(/Testing Mode Active/)).toBeInTheDocument();
    });

    it('displays smart contract requirements', () => {
      render(<TestingDashboard />);

      expect(screen.getByText('Smart Contract Requirements:')).toBeInTheDocument();
      expect(screen.getByText(/FEC Individual Limit.*\$3,300 per election/)).toBeInTheDocument();
      expect(screen.getByText(/Minimum.*\$1/)).toBeInTheDocument();
      expect(screen.getByText(/Complete Info.*All fields required/)).toBeInTheDocument();
      expect(screen.getByText(/Compliance.*FEC acknowledgment checkbox/)).toBeInTheDocument();
    });

    it('renders all form fields correctly', () => {
      render(<TestingDashboard />);

      expect(screen.getByLabelText(/Contribution Amount/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Wallet Address/)).toBeInTheDocument();
      expect(screen.getByLabelText(/First Name/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Last Name/)).toBeInTheDocument();
      expect(screen.getByLabelText(/Email/)).toBeInTheDocument();
      expect(screen.getByRole('checkbox')).toBeInTheDocument();
    });

    it('shows test button and empty results section', () => {
      render(<TestingDashboard />);

      expect(screen.getByText('🧪 Test Smart Contract Limits')).toBeInTheDocument();
      expect(screen.getByText(/No tests run yet/)).toBeInTheDocument();
    });
  });

  describe('Form Input Handling', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
      render(<TestingDashboard />);
    });

    it('allows user to input contribution amount', async () => {
      const amountInput = screen.getByLabelText(/Contribution Amount/);
      await user.type(amountInput, '100');

      expect(amountInput.value).toBe('100');
    });

    it('allows user to input wallet address', async () => {
      const walletInput = screen.getByLabelText(/Wallet Address/);
      await user.type(walletInput, '0x742d35cc6e4c43b3b5132b6c7e5c85b4a6b8b6a8');

      expect(walletInput.value).toBe('0x742d35cc6e4c43b3b5132b6c7e5c85b4a6b8b6a8');
    });

    it('allows user to input name fields', async () => {
      const firstNameInput = screen.getByLabelText(/First Name/);
      const lastNameInput = screen.getByLabelText(/Last Name/);

      await user.type(firstNameInput, 'John');
      await user.type(lastNameInput, 'Doe');

      expect(firstNameInput.value).toBe('John');
      expect(lastNameInput.value).toBe('Doe');
    });

    it('allows user to input email', async () => {
      const emailInput = screen.getByLabelText(/Email/);
      await user.type(emailInput, 'john@example.com');

      expect(emailInput.value).toBe('john@example.com');
    });

    it('allows user to toggle acknowledgment checkbox', async () => {
      const checkbox = screen.getByRole('checkbox');
      expect(checkbox).not.toBeChecked();

      await user.click(checkbox);
      expect(checkbox).toBeChecked();

      await user.click(checkbox);
      expect(checkbox).not.toBeChecked();
    });
  });

  describe('Form Validation', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
      render(<TestingDashboard />);
    });

    it('shows alert when required fields are missing', async () => {
      const alertSpy = vi.spyOn(window, 'alert').mockImplementation(() => {});

      const testButton = screen.getByText('🧪 Test Smart Contract Limits');
      await user.click(testButton);

      expect(alertSpy).toHaveBeenCalledWith('Please fill in all required fields');
      alertSpy.mockRestore();
    });

    it('does not run test when fields are incomplete', async () => {
      const testButton = screen.getByText('🧪 Test Smart Contract Limits');

      // Only fill some fields
      await user.type(screen.getByLabelText(/Contribution Amount/), '100');
      await user.type(screen.getByLabelText(/First Name/), 'John');

      await user.click(testButton);

      expect(processContribution).not.toHaveBeenCalled();
    });
  });

  describe('Smart Contract Testing', () => {
    let user;
    const fillAllFields = async (user) => {
      await user.type(screen.getByLabelText(/Contribution Amount/), '100');
      await user.type(
        screen.getByLabelText(/Wallet Address/),
        '0x742d35cc6e4c43b3b5132b6c7e5c85b4a6b8b6a8'
      );
      await user.type(screen.getByLabelText(/First Name/), 'John');
      await user.type(screen.getByLabelText(/Last Name/), 'Doe');
      await user.type(screen.getByLabelText(/Email/), 'john@example.com');
      await user.click(screen.getByRole('checkbox'));
    };

    beforeEach(() => {
      user = userEvent.setup();
      render(<TestingDashboard />);
    });

    it('runs successful smart contract test', async () => {
      await fillAllFields(user);

      const testButton = screen.getByText('🧪 Test Smart Contract Limits');
      await user.click(testButton);

      // Check loading state
      expect(screen.getByText('⏳ Testing Smart Contract...')).toBeInTheDocument();

      // Wait for completion
      await waitFor(() => {
        expect(processContribution).toHaveBeenCalledWith(
          {
            amount: '100',
            walletAddress: '0x742d35cc6e4c43b3b5132b6c7e5c85b4a6b8b6a8',
            firstName: 'John',
            lastName: 'Doe',
            email: 'john@example.com',
            acknowledgmentSigned: true,
          },
          'test-campaign-123'
        );
      });

      // Check success result display
      await waitFor(() => {
        expect(screen.getByText('✅ SUCCESS')).toBeInTheDocument();
        expect(screen.getByText(/Amount Tested.*\$100/)).toBeInTheDocument();
        expect(screen.getByText(/Name.*John Doe/)).toBeInTheDocument();
        expect(screen.getByText(/Email.*john@example.com/)).toBeInTheDocument();
        expect(screen.getByText(/Transaction Hash.*0xabc123/)).toBeInTheDocument();
        expect(screen.getByText(/Block.*18500000/)).toBeInTheDocument();
        expect(screen.getByText(/Gas Used.*21000/)).toBeInTheDocument();
      });
    });

    it('handles smart contract test failure', async () => {
      processContribution.mockResolvedValue({
        success: false,
        error: 'Amount exceeds FEC limit',
      });

      await fillAllFields(user);

      const testButton = screen.getByText('🧪 Test Smart Contract Limits');
      await user.click(testButton);

      await waitFor(() => {
        expect(screen.getByText('❌ FAILED')).toBeInTheDocument();
        expect(screen.getByText(/Error.*Amount exceeds FEC limit/)).toBeInTheDocument();
      });
    });

    it('handles smart contract test exception', async () => {
      processContribution.mockRejectedValue(new Error('Network connection failed'));

      await fillAllFields(user);

      const testButton = screen.getByText('🧪 Test Smart Contract Limits');
      await user.click(testButton);

      await waitFor(() => {
        expect(screen.getByText('❌ FAILED')).toBeInTheDocument();
        expect(screen.getByText(/Error.*Network connection failed/)).toBeInTheDocument();
      });
    });

    it('updates button state during testing', async () => {
      processContribution.mockImplementation(
        () => new Promise((resolve) => setTimeout(() => resolve({ success: true }), 100))
      );

      await fillAllFields(user);

      const testButton = screen.getByText('🧪 Test Smart Contract Limits');
      await user.click(testButton);

      // Check loading state
      expect(screen.getByText('⏳ Testing Smart Contract...')).toBeInTheDocument();
      expect(testButton).toBeDisabled();

      // Wait for completion
      await waitFor(() => {
        expect(screen.getByText('🧪 Test Smart Contract Limits')).toBeInTheDocument();
      });
      expect(testButton).not.toBeDisabled();
    });

    it('displays multiple test results in chronological order', async () => {
      // First test
      await fillAllFields(user);
      await user.click(screen.getByText('🧪 Test Smart Contract Limits'));

      await waitFor(() => {
        expect(screen.getByText('✅ SUCCESS')).toBeInTheDocument();
      });

      // Second test with different data
      await user.clear(screen.getByLabelText(/Contribution Amount/));
      await user.type(screen.getByLabelText(/Contribution Amount/), '250');

      processContribution.mockResolvedValue({
        success: false,
        error: 'Test error',
      });

      await user.click(screen.getByText('🧪 Test Smart Contract Limits'));

      await waitFor(() => {
        expect(screen.getAllByText(/SUCCESS|FAILED/)).toHaveLength(2);
        expect(screen.getByText('❌ FAILED')).toBeInTheDocument();
        expect(screen.getByText(/Amount Tested.*\$250/)).toBeInTheDocument();
      });
    });
  });

  describe('Accessibility', () => {
    it('has proper form labels and ARIA attributes', () => {
      render(<TestingDashboard />);

      const amountInput = screen.getByLabelText(/Contribution Amount/);
      const walletInput = screen.getByLabelText(/Wallet Address/);
      const firstNameInput = screen.getByLabelText(/First Name/);
      const lastNameInput = screen.getByLabelText(/Last Name/);
      const emailInput = screen.getByLabelText(/Email/);

      expect(amountInput).toBeInTheDocument();
      expect(walletInput).toBeInTheDocument();
      expect(firstNameInput).toBeInTheDocument();
      expect(lastNameInput).toBeInTheDocument();
      expect(emailInput).toBeInTheDocument();

      // Check input types
      expect(amountInput).toHaveAttribute('type', 'number');
      expect(emailInput).toHaveAttribute('type', 'email');
    });

    it('has proper button roles and accessible text', () => {
      render(<TestingDashboard />);

      const testButton = screen.getByRole('button', { name: /Test Smart Contract Limits/ });
      expect(testButton).toBeInTheDocument();
    });

    it('maintains focus management', async () => {
      const user = userEvent.setup();
      render(<TestingDashboard />);

      const amountInput = screen.getByLabelText(/Contribution Amount/);
      amountInput.focus();

      await user.type(amountInput, '100');
      expect(amountInput).toHaveFocus();
    });
  });

  describe('Component Snapshots', () => {
    it('matches snapshot for initial state', () => {
      const { container } = render(<TestingDashboard />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot with test results', async () => {
      const user = userEvent.setup();
      render(<TestingDashboard />);

      // Fill form and run test
      await user.type(screen.getByLabelText(/Contribution Amount/), '100');
      await user.type(
        screen.getByLabelText(/Wallet Address/),
        '0x742d35cc6e4c43b3b5132b6c7e5c85b4a6b8b6a8'
      );
      await user.type(screen.getByLabelText(/First Name/), 'John');
      await user.type(screen.getByLabelText(/Last Name/), 'Doe');
      await user.type(screen.getByLabelText(/Email/), 'john@example.com');
      await user.click(screen.getByRole('checkbox'));

      await user.click(screen.getByText('🧪 Test Smart Contract Limits'));

      await waitFor(() => {
        expect(screen.getByText('✅ SUCCESS')).toBeInTheDocument();
      });

      const { container } = render(<TestingDashboard />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
