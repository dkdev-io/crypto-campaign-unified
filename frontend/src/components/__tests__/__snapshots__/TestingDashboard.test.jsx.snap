// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestingDashboard > Component Snapshots > matches snapshot for initial state 1`] = `
<div
  style="padding: 2rem; max-width: 1200px; margin: 0px auto;"
>
  <div
    style="text-align: center; margin-bottom: 2rem;"
  >
    <h1
      style="color: rgb(42, 42, 114);"
    >
      ðŸ§ª Smart Contract Testing Dashboard
    </h1>
    <p
      style="color: rgb(102, 102, 102); font-size: 1.1rem;"
    >
      Test contribution limits without triggering wallet extensions
    </p>
    <div
      style="background: rgb(227, 242, 253); padding: 1rem; border-radius: 8px; margin: 1rem 0px; border: 1px solid rgb(187, 222, 251);"
    >
      <strong>
        Testing Mode Active:
      </strong>
       No wallet extensions will be triggered. All transactions are simulated.
    </div>
  </div>
  <div
    style="display: grid; grid-template-columns: 1fr 1fr; gap: 2rem;"
  >
    <div>
      <h2
        style="margin-bottom: 1rem;"
      >
        Smart Contract Test
      </h2>
      <div
        style="background: rgb(248, 249, 250); padding: 1rem; border-radius: 8px; margin-bottom: 1rem;"
      >
        <h3>
          Smart Contract Requirements:
        </h3>
        <ul>
          <li>
            <strong>
              FEC Individual Limit:
            </strong>
             $3,300 per election (cumulative)
          </li>
          <li>
            <strong>
              Minimum:
            </strong>
             $1
          </li>
          <li>
            <strong>
              Complete Info:
            </strong>
             All fields required
          </li>
          <li>
            <strong>
              Compliance:
            </strong>
             FEC acknowledgment checkbox must be signed
          </li>
          <li>
            <strong>
              KYC:
            </strong>
             Requirements (to be added later)
          </li>
        </ul>
      </div>
      <div
        style="display: grid; gap: 1rem; margin-bottom: 1rem;"
      >
        <div>
          <label
            style="display: block; margin-bottom: 0.5rem; font-weight: bold;"
          >
            Contribution Amount *
          </label>
          <input
            min="0"
            placeholder="Enter amount to test"
            step="0.01"
            style="width: 100%; padding: 0.75rem; border: 1px solid rgb(204, 204, 204); border-radius: 4px;"
            type="number"
            value=""
          />
        </div>
        <div>
          <label
            style="display: block; margin-bottom: 0.5rem; font-weight: bold;"
          >
            Wallet Address *
          </label>
          <input
            placeholder="Enter wallet address"
            style="width: 100%; padding: 0.75rem; border: 1px solid rgb(204, 204, 204); border-radius: 4px;"
            type="text"
            value=""
          />
        </div>
        <div
          style="display: grid; grid-template-columns: 1fr 1fr; gap: 1rem;"
        >
          <div>
            <label
              style="display: block; margin-bottom: 0.5rem; font-weight: bold;"
            >
              First Name *
            </label>
            <input
              placeholder="First name"
              style="width: 100%; padding: 0.75rem; border: 1px solid rgb(204, 204, 204); border-radius: 4px;"
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              style="display: block; margin-bottom: 0.5rem; font-weight: bold;"
            >
              Last Name *
            </label>
            <input
              placeholder="Last name"
              style="width: 100%; padding: 0.75rem; border: 1px solid rgb(204, 204, 204); border-radius: 4px;"
              type="text"
              value=""
            />
          </div>
        </div>
        <div>
          <label
            style="display: block; margin-bottom: 0.5rem; font-weight: bold;"
          >
            Email *
          </label>
          <input
            placeholder="Email address"
            style="width: 100%; padding: 0.75rem; border: 1px solid rgb(204, 204, 204); border-radius: 4px;"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        style="background: rgb(240, 248, 255); padding: 1rem; border-radius: 4px; margin-bottom: 1rem; font-size: 0.9rem;"
      >
        <label
          style="display: flex; align-items: flex-start; gap: 0.5rem;"
        >
          <input
            style="margin-top: 0.25rem;"
            type="checkbox"
          />
          <span>
            I certify that I am a U.S. citizen or lawfully admitted permanent resident, this contribution is made from my own funds, I am not a federal contractor, and I am at least 18 years old.
          </span>
        </label>
      </div>
      <button
        style="width: 100%; padding: 1rem; background: rgb(111, 66, 193); color: white; border-radius: 8px; font-size: 1.1rem; cursor: pointer;"
      >
        ðŸ§ª Test Smart Contract Limits
      </button>
    </div>
    <div>
      <h2
        style="margin-bottom: 1rem;"
      >
        Test Results
      </h2>
      <div
        style="display: grid; gap: 1rem; max-height: 600px; overflow-y: auto;"
      >
        <div
          style="background: rgb(248, 249, 250); padding: 2rem; text-align: center; border-radius: 8px; color: rgb(102, 102, 102);"
        >
          No tests run yet. Fill in the form and click "Test Smart Contract Limits" to test your data.
        </div>
      </div>
    </div>
  </div>
</div>
`;
